client/src/components/campaignCreatorComponents/TriggerEventsSettings.tsx

```diff
@@ -147,6 +146,32 @@ const CampaignFinalSettings = ({
             }, {});
     };
 
+    const createGearGroup = (categoryKey: keyof TypeLocationsData) => {
+        console.log('cat key', categoryKey);
+        return locationsData[categoryKey].gear.length === 0
+            ? {}
+            : locationsData[categoryKey].gear.reduce((currentLocationObject, gear) => {
+                  return {
+                      ...currentLocationObject,
+                      [gear]: 0,
+                  };
+              }, {});
+    };
```

Comment:
I don't think you need the `locationsData[categoryKey].gear.length === 0
            ? {}` at the top.  I believe the reduce will return `{}` if `gear` is empty.

Reply:
i was testing something, forgot to remove